### PR TITLE
Add host cpu info

### PR DESCRIPTION
**Pull request by ChrsMark**
This PR adds CPU information as proposed with https://github.com/open-telemetry/semantic-conventions/issues/130.

The CPU related information go under the `host.cpu` namespace.

### CODE REVIEW

File: model/resource/host.yaml
Comment (mx-psi) [inline model/resource/host.yaml:83]: Should we have a fixed list of vendor IDs? Similar to what we have for e.g. `host.arch`

Comment (ChrsMark) [reply model/resource/host.yaml:83]: That's a good idea but from what I can see the list of known processor manufacturer ID strings can be quite big. For example I find a list of vendors at https://en.m.wikipedia.org/wiki/CPUID#EAX=0:_Highest_Function_Parameter_and_Manufacturer_ID. I guess it's a slightly more "flexible" field than the `arch` one, right?

File: docs/resource/host.md
Comment (mx-psi) [inline docs/resource/host.md:45]: Based on #99 I think these would probably make more sense as a metric (at least if these can change). @frzifus any thoughts about this?

Comment (frzifus) [reply docs/resource/host.md:45]: From my understanding the cache size even in a vm highly depends on the hardware, right? Means the only way to change the cache size would be to swap the CPU. As long as those attribute values are associated with a specific cpu, that should not matter.

Thats different for the CPU frequency. On a normal linux machine it can be changed by writting a number to `/sys/..`. (see [scaling_setspeed](https://superuser.com/questions/533507/how-can-i-manually-set-the-cpu-frequency)) and the other way around for example aws instances [offer a boosted frequency](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/processor_state_control.html).

Comment (ChrsMark) [reply docs/resource/host.md:45]: Yes to my findings this one (coming from `/proc/cpuinfo`) refers to the L2 cache of the CPU, which is fixed. 
For reference LLC or L3 cache can be allocated dynamically in some systems but we don't refer to this one with this attribute.

For the frequency it's weird to me that the value is changing since I would expect it to be fixed to the model's value and not the actual freq of the cores'. However, I did some tests and it's actually changing so it's seems that I'm wrong here.

**Comment by ChrsMark** [comment]
Hey @mx-psi @frzifus ! 

I have moved the Frequency information out of this patch and only kept the resource attributes. I think we can unblock this one and add the Frequency as a metric later after we have https://github.com/open-telemetry/semantic-conventions/pull/89 merged and https://github.com/open-telemetry/semantic-conventions/issues/226 can give more clarity about metrics' namespace.
Let me know what you think :).

**Review by frzifus — APPROVED** [review]
LGTM, ~should we then add the frequency as a metric afterwards?~

Missed ([comment](https://github.com/open-telemetry/semantic-conventions/pull/209#issuecomment-1665176101)). :smile:

**Comment by ChrsMark** [comment]
> LGTM, ~should we then add the frequency as a metric afterwards?~
> 
> Missed ([comment](https://github.com/open-telemetry/semantic-conventions/pull/209#issuecomment-1665176101)). smile

:) Yeap I think we should treat it as a Metric. example:

```console
➜  semantic-conventions git:(add_host_cpu_info) cat /proc/cpuinfo | grep MHz                                    
cpu MHz		: 3000.000
cpu MHz		: 1054.524
cpu MHz		: 1093.029
cpu MHz		: 3000.000
cpu MHz		: 3000.000
cpu MHz		: 3000.000
cpu MHz		: 3000.000
cpu MHz		: 3000.000
➜  semantic-conventions git:(add_host_cpu_info) cat /proc/cpuinfo | grep MHz
cpu MHz		: 1467.399
cpu MHz		: 3000.000
cpu MHz		: 3000.000
cpu MHz		: 3000.000
cpu MHz		: 3000.000
cpu MHz		: 3000.000
cpu MHz		: 2462.027
cpu MHz		: 3000.000
```

File: model/resource/host.yaml
Comment (mx-psi) [reply model/resource/host.yaml]: I guess since these are fixed this is fine we can mark this as solved. 

My aim was to make it the language a bit more understandable i.e. using Intel instead of "GenuineIntel" or AMD instead of "AuthenticAMD". But it seems like it is common enough to 'fake' the CPUID that we can just leave the value without 'interpreting' it.

File: docs/resource/host.md
Comment (mx-psi) [inline docs/resource/host.md:44]: I am still not familiar enough with the 'rules' we use for conventions naming, so I have a couple questions:

1. Should we have `host.cpu.model.id` and `host.cpu.model.name` instead of the current wording? Since they both seem to refer to properties of the model. `stepping` could also be part of the model, at least semantically.
2. Similarly, do we expect to ever report properties about other caches (other than L2) or properties other than the size? Should that make us change into... `cache.size`? `cache.l2.size`? It's still unclear to me what level of specificity we need in the naming (it looks like I tend to want to be more specific than most, based on the `host.ip` discussion).

Comment (ChrsMark) [reply docs/resource/host.md]: yeap, I had the same though about grouping those. I think it's more future proof to follow what you mention. I tried to apply this with [356e3f8](https://github.com/open-telemetry/semantic-conventions/pull/209/commits/356e3f8a293cad107ac1ba3c47df0d531562d2dd).

**Review by mx-psi — APPROVED** [review]

File: model/resource/host.yaml
Comment (AlexanderWert) [inline model/resource/host.yaml:83]: Where does this limitation of 12-character string come from?
If that's a specific format, can we link to the specification from here?

File: docs/resource/host.md
Comment (AlexanderWert) [reply docs/resource/host.md]: Agree with the `host.cpu.model.*` as a namespace.
But what about `host.cpu.vendor.id`? Do we expect other attributes under `host.cpu.vendor.*`? If not maybe it's better to leave it as `host.cpu.vendor_id`?

File: model/resource/host.yaml
Comment (ChrsMark) [reply model/resource/host.yaml:83]: From what I found it comes from the fact that it is stored directly in 3 specific CPU's registers: https://en.wikipedia.org/wiki/CPUID#EAX=0:_Highest_Function_Parameter_and_Manufacturer_ID. So 3x4bytes=12 bytes=12 ASCII characters.

This information is coming directly from the [kernel](https://www.kernel.org/doc/Documentation/filesystems/proc.txt) and unfortunately I cannot find any well defined section for it within the kernel docs: https://www.kernel.org/doc/html/latest/search.html?q=cpuinfo. It seems that some docs where added recently for cpuinfo flags but this does not help a lot here: https://lwn.net/ml/linux-kernel/20200610200701.16757-1-kyung.min.park@intel.com/.

I will look again though in case I can find something more specific.

File: docs/resource/host.md
Comment (ChrsMark) [reply docs/resource/host.md]: For `cache` maybe in the future there will be need to store the cache's blocks or cache model name. Also other levels of cache might be available as well `L2`, `L3` etc.

For vendor ID I think it's most specific like this and we might need the Vendor info directly in the future. For example there are `AMDisbetter!` and `AuthenticAMD` as possible vendor_id values but both come from `AMD`, the only difference it's their historical date.

In any case I don't have a strong preference here, so I would be ok to change those if we don't see a lot of value.

**Comment by mx-psi** [comment]
@AlexanderWert mind taking another look?

File: model/resource/host.yaml
Comment (AlexanderWert) [reply model/resource/host.yaml:83]: @ChrsMark Maybe we could link this one: https://wiki.osdev.org/CPUID ?

File: docs/resource/host.md
Comment (joaopgrassi) [inline docs/resource/host.md:44]: Is the `Opt-In` a deliberate choice these new attributes? Do you think it aligns with the description of Opt-in we have? 
https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/attribute-requirement-level.md#opt-in

**Review by joaopgrassi — COMMENTED** [review]
Please add a changelog entry

File: docs/resource/host.md
Comment (ChrsMark) [reply docs/resource/host.md:44]: I would consider those as more "advanced" attributes that might not be of importance for all use-cases, that's why I went for the safe Opt-In for now.
I would be ok to change it though.
@mx-psi @frzifus any thoughts on this?

**Review by AlexanderWert — APPROVED** [review]

File: docs/resource/host.md
Comment (joaopgrassi) [reply docs/resource/host.md:44]: > I would consider those as more "advanced" attributes that might not be of importance for all use-cases, that's why I went for the safe Opt-In for now.

I think so to. The reason I'm asking is because for Opt-In, the "instrumentation" would need to offer a way for users to opt-in into these attributes explicitly. Is something like this possible in the collector today?

Comment (joaopgrassi) [reply docs/resource/host.md:44]: Perfect!

**Review by mx-psi — APPROVED** [review]

**Review by joaopgrassi — APPROVED** [review]

**Comment by joaopgrassi** [comment]
@ChrsMark make sure to resolve the discussions if you think they are resolved. That would help others to see if this is in a final state or not.

File: model/resource/host.yaml
Comment (ChrsMark) [reply model/resource/host.yaml:83]: Added. Resolving this as complete.

File: docs/resource/host.md
Comment (ChrsMark) [reply docs/resource/host.md]: Frequency will be added as metric.

File: docs/resource/host.md
Comment (ChrsMark) [reply docs/resource/host.md]: Marking as solved since the current changes are already approved.

**Comment by ChrsMark** [comment]
Thanks @joaopgrassi, based on the latest discussions/approvals all those are resolved. Marked them properly.

**Review by frzifus — APPROVED** [review]

**Comment by ChrsMark** [comment]
@joaopgrassi @AlexanderWert anything else missing here?

**Comment by joaopgrassi** [comment]
@open-telemetry/specs-semconv-approvers @open-telemetry/specs-semconv-maintainers we discussed this in the System semconv SIG and should be good to go. Will give it another day, but if nothing then I will merge.

**Pull request merged**
